docs/Scripts.md

```diff
@@ -0,0 +1,122 @@
+# Lua Scripts
+
+Dolphin ships support for Lua 5.3 scripts that run embedded with the emulator.
+
+*Caution:* Never run untrusted scripts or scripts you do not understand yourself!
+Scripts will have access to your system like any other program, including the ability to do nasty things.
+
+**Requirements**
+
+Script support is disabled by default, and has to be enabled via CMake flag `-DLUA_SCRIPTS=true`.
+
+Scripting support requires Lua sources while compiling and the Lua 5.3 dynamic library at runtime.
+Debian provides both with the `liblua5.3-dev` and `lua5.3` packages.
+
+**Usage**
+
+Scripts are defined in game properties and are run just before emulation starts.
+Any Lua code blocks emulator progression.
+
+## `dolphin` API reference
+
+**In development:** There are no stability guarantees. The scripting API will change very likely in future versions of Dolphin.
+
+Lua scripts use the `dolphin` module for interacting with the emulator.
+
+### PowerPC state (`ppc.*`)
+
+Inspect and manipulate PowerPC register state.
+
+#### Registers
+
+| identifier              | type             | description                        |
+| ----------------------- | ---------------- | ---------------------------------- |
+| `dolphin.ppc.gpr[x]`    | `u32[32]`        | GPRs (general purpose registers)   |
+| `dolphin.ppc.pc`        | `u32`            | PC (program counter)               |
+| `dolphin.ppc.npc`       | `u32`            | NPC (next program counter)         |
+| `dolphin.ppc.sr[x]`     | `u32[16]`        | SRs (segment registers)            |
+| `dolphin.ppc.spr[x]`    | `u32[1024]`      | SPRs (special purpose registers)   |
+| `dolphin.ppc.ps_u32[x]` | `(u32, u32)[32]` | paired single registers (u32 mode) |
+| `dolphin.ppc.ps_u64[x]` | `(u64, u64)[32]` | paired single registers (u64 mode) |
+| `dolphin.ppc.ps_f64[x]` | `(f64, f64)[32]` | paired single registers (f64 mode) |
+
+### Memory access
+
+Access Wii memory from the main processor's view.
+
+#### Memory arrays
+
+Lua tables taking an address as index. Supports unaligned access.
+
+| identifier            | type     | description               |
+| --------------------- | -------- | ------------------------- |
+| `dolphin.ppc.mem_u8`  | `u8[*]`  | Read/write u8             |
```

Comment:
These aren't part of ppc in the implementation, which seems correct. Copy/paste bug in the docs?